src/PerfView/PerfViewData.cs

```diff
@@ -3388,6 +3388,436 @@ private void CreateCSVFile(List<AspNetRequest> requests, string csvFileName)
+        private Dictionary<string, ANCHostingRequest> incompleteRequests = new Dictionary<string, ANCHostingRequest>();
+        private List<ANCHostingRequest> completeRequests = new List<ANCHostingRequest>();
+        private HashSet<ProcessIndex> processesThatEmittedEvents = new HashSet<ProcessIndex>();
+
+        // holds non-request events like host/start and stop, etc.
+        private List<OtherHostingEvent> otherEvents = new List<OtherHostingEvent>();
+
+        public PerfViewAspNetCoreStats(PerfViewFile dataFile) : base(dataFile, "ASP.NET Core Stats") { }
+
+        protected override void WriteHtmlBody(TraceLog dataFile, TextWriter outputWriter, string fileName, TextWriter log)
+        {
+            outputWriter.WriteLine("<H2>ASP.NET Core Request Statistics</H2>");
+
+            // request/stop does not contain status code or anything like that
+            // maybe add ILogger events, if available, to enhance these logs?
+            // possibility: the ActivityStart/Stop events in System.Diagnostics.DiagnosticSource for ANCHosting do contain status code, check for these?
+            // also down the road if both ILogger and MS.ANC.Hosting events are available, use both and integrate them?
+            /* 
+             * some possible events (ref: https://source.dot.net/#Microsoft.AspNetCore.Hosting/Internal/HostingEventSource.cs):
+             * Request/Start and Request/Stop
+             * ManifestData
+             * Host/Start and Host/Stop
+             * UnhandledException
+             * ServerReady
+             */
+
+            // using the push/callback model for TraceLog processing
+            var dispatcher = dataFile.Events.GetSource();
+
+            // for inbound requests/sec stats, use an int counter
+            int inboundRequestCount = 0;
+
+            // this callback is invoked when the dispatcher.Process() call is made later, upon matching ANCHosting provider events (any of them)
+            dispatcher.Dynamic.AddCallbackForProviderEvent(_hostingProvider, null, delegate (TraceEvent traceEvent)
+            {
+                // track which processes (via the trace's ProcessIndex) have emitted events - this is used for output in the stats later
+                _ = processesThatEmittedEvents.Add(traceEvent.Process().ProcessIndex);
+
+                ANCHostingRequest request = null;
+                // determine which event we're dealing with
+                switch (traceEvent.EventName)
+                {
+                    case _eventRequestStart:
+                        inboundRequestCount++;
+                        if (incompleteRequests.TryGetValue(ANCHostingRequest.GetIndexingKeyFromEvent(traceEvent), out request))
+                        {
+                            // request exists, update its properties
+                            request.StartTimeRelativeMSec = traceEvent.TimeStampRelativeMSec;
+                            request.HasStart = true;
```

Comment:
I think other than right after construction `request.HashStart` always equals `true`.  I think (but do check my work), that you don't need this property.

Reply:
HasStart and HasStop are both set to _false_ on construction -- they are individually set to _true_ when the corresponding event is found in the TraceLog. This is one area that I did change in the recent commit to help with keeping track of timing vs. event-related completeness. HasStart and HasStop used to return _true_ if the request had a nonzero start or stop time. Now, since I set start/stop time manually for incomplete requests (for timing purposes), I want to ensure that HasStart/HasStop are only true IF an actual start or stop event were encountered. This is then used to help the user see which scenario an incomplete request is.

Reply:
What you're describing makes sense to me, but I see `request.HasStart = true` being set when a request stop event shows up regardless of whether or not there is an incomplete request.  That's where my thought that this isn't necessary comes from.  Is that just maybe unintentional?  I wouldn't expect to see this when a stop event shows up.

Reply:
Hey Brian, unless I missed it, request.HasStart is only set to _true_ in 2 places, both inside of the code for processing a request/start event. In the request/stop event, only the HasStop is set to true:
![image](https://github.com/microsoft/perfview/assets/38960254/577248e6-71be-40db-a552-da952af573da)
Let me know if I missed anything there.

Reply:
You're right - looking at this again, I do not see this issue anymore (or I just completely mis-read it).